abi.go

```diff
@@ -0,0 +1,170 @@
+package ebpf
+
+import (
+	"github.com/pkg/errors"
+)
+
+// CollectionABI describes the interface of an eBPF collection.
+type CollectionABI struct {
+	Maps     map[string]*MapABI
+	Programs map[string]*ProgramABI
+}
+
+// CheckSpec verifies that all maps and programs mentioned
+// in the ABI are present in the spec.
+func (abi *CollectionABI) CheckSpec(cs *CollectionSpec) error {
```

Comment:
Why can't we check the Spec conforms to the ABI? AFAICT all the fields used in `progABI.Check()` and `mapABI.Check()` are in the corresponding `Spec`s

Reply:
We can, it's just not that useful. If someone is working with a spec they'll probably want to change parts of the program, otherwise they'd just get a Collection in the first place. So checking CollectionSpec isn't enough, and we'd just re-check when the collection is created.

From my POV the ABI is only useful once dealing with instantiated Maps and Program, because that's what the code will make assumptions about.